source/extensions/omni.isaac.lab/omni/isaac/lab/envs/mdp/observations.py

```diff
@@ -347,61 +375,137 @@ def __call__(
+            """
+            # Move the image to the model device
+            img = img.to(model_device)
+            # Normalize the image
+            min_img = torch.amin(img, dim=(1, 2), keepdim=True)
+            max_img = torch.amax(img, dim=(1, 2), keepdim=True)
+
+            return (img - min_img) / (max_img - min_img)
+
+        def _inference(model, images: torch.Tensor) -> torch.Tensor:
+            """Inference the Theia transformer model.
+
+            Args:
+                model: The Theia transformer model.
+                images: The preprocessed image tensor. Shape is (num_envs, height, width, channel).
+
+            Returns:
+                The extracted features tensor. Shape is (num_envs, feature_dim).
+            """
+            return model.forward_feature(images, do_rescale=False, interpolate_pos_encoding=True)
 
-        image_device = images.device
+        # return the model, preprocess and inference functions
+        return {
+            "model": _load_model,
+            "preprocess": _preprocess_image,
+            "inference": _inference,
+        }
 
-        if model_device is not None:
-            images = images.to(model_device)
+    def _prepare_resnet_model(self, model_name: str, model_device: str) -> dict:
+        """Prepare the ResNet model for inference.
 
-        proc_images = self.model_zoo_cfg[model_name]["preprocess"](images)
-        features = self.model_zoo_cfg[model_name]["inference"](self.model_zoo[model_name], proc_images)
+        Args:
+            model_name: The name of the ResNet model to prepare.
+            model_device: The device to store and infer the model on.
+
+        Returns:
+            A dictionary containing the model, preprocess and inference functions.
+        """
+        from torchvision import models
 
-        return features.to(image_device).clone()
+        def _load_model() -> torch.nn.Module:
+            """Load the ResNet model."""
+            model = getattr(models, model_name)(pretrained=True).eval()
+            return model.to(model_device)
+
+        def _preprocess_image(img: torch.Tensor) -> torch.Tensor:
+            """Preprocess the image for the ResNet model.
+
+            Args:
+                img: The image tensor to preprocess. Shape is (num_envs, height, width, channel).
+
+            Returns:
+                The preprocessed image tensor. Shape is (num_envs, channel, height, width).
+            """
+            # move the image to the model device
+            img = img.to(model_device)
+            # permute the image to (num_envs, channel, height, width)
+            img = img.permute(0, 3, 1, 2)
+            # normalize the image
+            mean = torch.tensor([0.485, 0.456, 0.406], device=img.device).view(1, 3, 1, 1)
+            std = torch.tensor([0.229, 0.224, 0.225], device=img.device).view(1, 3, 1, 1)
+
+            return (img - mean) / std
```

Comment:
"preprocess": (
                    lambda img: (
                        (
                            img.permute(
                                0, 3, 1, 2
                            ).float()  # Convert [batch, height, width, 3] -> [batch, 3, height, width]
                            - torch.tensor([0.485, 0.456, 0.406], device=img.device).view(1, 3, 1, 1)
                        )
                        / torch.tensor([0.229, 0.224, 0.225], device=img.device).view(1, 3, 1, 1)
                    )
                    / 255 # need to normalize this way
                ),